server/routes/store.py

```diff
@@ -1,8 +1,49 @@
-from flask import Blueprint
-from server.models import Store
+import json
+
+from flask import Blueprint, session, request
+from sqlalchemy.sql import func
+from server.models import Store, Transaction, Item
+from server import db
 
 store_bp = Blueprint("store_bp", __name__, url_prefix="/api/store")
 
+def valid_funds(user_id, item_type):
+    total_tickets = (
+        db.session.query(func.sum(Transaction.ticket_amount))\
+        .filter(Transaction.user_id == user_id)
+        .scalar()
+    )
+    item_price = Store.query.filter_by(item_type=item_type).first().price
+    if total_tickets is None or item_price is None:
+        return False
+    return total_tickets >= item_price
+
+def item_available(user_id, item_type, quantity):
+    item_info = Store.query.filter_by(item_type=item_type).first()
+    item_query = Item.query.filter_by(user_id=user_id, item_type=item_type).first()
+    if item_query is not None and item_query.count + quantity > item_info.limit:
+        return False
+    return True
+
+def make_purchase(user_id, item_type, quantity):
+    item_info = Store.query.filter_by(id=item_type).first()
+    transaction = Transaction(
+        user_id=user_id,
+        ticket_amount=item_info.price,
+        activity="ticketstore",
+    )
+    db.session.add(transaction)
+    item_query = Item.query.filter_by(user_id=user_id, item_type=item_type).first()
+    if item_query is None:
+        item = Item(
+            item_type = item_type,
```

Comment:
item_type = id

Reply:
why id?

Reply:
id in store is what item_type should've been.